analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/ExecutingSqlQueries.cs

```diff
@@ -81,10 +82,56 @@ public ExecutingSqlQueries()
             && expression is BinaryExpressionSyntax concatenation
             && !IsConcatenationOfConstants(concatenation, semanticModel);
 
-        private bool IsTrackedVariableDeclaration(ExpressionSyntax argument, SemanticModel semanticModel) =>
-            (argument is IdentifierNameSyntax identifierNameSyntax
-             && semanticModel.GetDeclaringSyntaxNode(identifierNameSyntax) is VariableDeclaratorSyntax variableDeclaratorSyntax
-             && IsTracked(variableDeclaratorSyntax.Initializer?.Value, semanticModel));
+        private static bool IsTrackedVariableDeclaration(ExpressionSyntax expression, BaseContext context)
+        {
+            if (expression is IdentifierNameSyntax)
+            {
+                var node = expression;
+                Location location;
+                do
+                {
+                    var identifierName = (node as IdentifierNameSyntax).Identifier.ValueText;
+                    node = AssignmentFinder.FindLinearPrecedingAssignmentExpression(identifierName, node) as ExpressionSyntax;
+                    location = GetSecondaryLocationForExpression(node);
+
+                    if (IsSensitiveExpression(node, context.SemanticModel))
+                    {
+                        if (location != Location.None)
+                        {
+                            context.AddSecondaryLocation(new SecondaryLocation(location, string.Format(AssignmentWithFormattingMessage, identifierName)));
+                        }
+
+                        return true;
+                    }
+                    else
+                    {
+                        if (location != Location.None)
+                        {
+                            context.AddSecondaryLocation(new SecondaryLocation(location, string.Format(AssignmentMessage, identifierName)));
+                        }
+                    }
+                }
+                while (node is IdentifierNameSyntax);
+            }
+
+            return false;
+        }
+
+        private static Location GetSecondaryLocationForExpression(ExpressionSyntax node)
```

Comment:
I don't like `Get` prefixes for many reasons (Costin disagrees), I'd use it only when there's no other choice and the method really only gets something, that directly exists.
```suggestion
        private static Location SecondaryLocationForExpression(ExpressionSyntax node)
```